lvsfunc/comparison.py

```diff
@@ -2,17 +2,347 @@
+
+        For example, for 3 clips, the automatic arrangement becomes:
+        [
+         [1, 1],
+         [1, 0]
+        ].
+
+        For 7 clips, the automatic arrangement becomes:
+        [
+         [1, 1, 1],
+         [1, 1, 1],
+         [1, 0, 0]
+        ].
+
+        For custom arrangements, such as (for 4 clips):
+        [
+         [0, 1, 0, 1],
+         [1],
+         [0, 1]
+        ],
+        the rows will be auto-padded with `0` to be the same length.
+
+    :param clips:           A dict mapping names to clips or simply a sequence of clips in a tuple or a list.
+                            If specified a dict, the names will be overlayed on the clips using
+                            ``vapoursynth.VideoNode.text.Text``.
+                            If given a simple sequence of clips, the `label_alignment` parameter will have no effect
+                            and the clips will not be labeled.
+                            The order of the clips in either a dict or a sequence will be kept in the comparison.
+    :param arrangement:     2-dimension array (list of lists) of 0s and 1s representing a list of rows of clips(1)
+                            or blank spaces(0) (Default: None)
+    :param label_alignment: An integer from 1-9, corresponding to the positions of the keys on a numpad.
+                            Only used if `clips` is a dict.
+                            Determines where to place clip name using ``vapoursynth.VideoNode.text.Text`` (Default: 7)
+    """
+
+    def __init__(self,
+                 clips: Union[Dict[str, vs.VideoNode], Sequence[vs.VideoNode]],
+                 /,
+                 *,
+                 arrangement: Optional[List[List[int]]] = None,
+                 label_alignment: int = 7
+                 ) -> None:
+        super().__init__(clips, label_alignment=label_alignment)
+
+        if not self.width or not self.height:
+            raise ValueError("Tile: all clip widths, and heights must be the same")
+
+        if arrangement is None:
+            self.arrangement = self._auto_arrangement()
+        else:
+            is_one_dim = len(arrangement) < 2 or all(len(row) == 1 for row in arrangement)
+            if is_one_dim:
+                raise ValueError("Tile: use Stack instead if the array is one dimensional")
+            self.arrangement = arrangement
+
+        self.blank_clip = core.std.BlankClip(clip=self.clips[0], keep=1)
+
+        if not all(len(row) == (max_length := max(len(row) for row in self.arrangement)) for row in self.arrangement):
+            for row in self.arrangement:
+                if len(row) != max_length:
+                    diff = max_length - len(row)
+                    for _ in range(diff):
+                        row.append(0)  # padding all rows to the same length
+
+        array_count = 0
+        for list_ in self.arrangement:
+            for elem in list_:
+                if elem:
+                    array_count += 1
```

Comment:
```suggestion
        array_count = sum(map(sum, self.arrangement))
```
since all cells are either `0` or `1`.